src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandRunner.cs

```diff
@@ -46,57 +46,114 @@ public static int ExecuteCommand(WhyCommandArgs whyCommandArgs)
+                    }
+                    else
+                    {
+                        whyCommandArgs.Logger.LogMinimal(
+                            string.Format(
+                                Strings.WhyCommand_Message_NoDependencyGraphsFoundInProject,
+                                assetsFile.PackageSpec.Name,
+                                targetPackage));
+                    }
+                }
+                else
+                {
+                    anyErrors = true;
+                }
+            }
 
-                        Dictionary<string, List<DependencyNode>?>? dependencyGraphPerFramework = DependencyGraphFinder.GetAllDependencyGraphsForTarget(
-                            assetsFile,
-                            whyCommandArgs.Package,
-                            whyCommandArgs.Frameworks);
+            return anyErrors ? ExitCodes.Error : ExitCodes.Success;
+        }
 
-                        if (dependencyGraphPerFramework != null)
+        private static IEnumerable<(string assetsFilepath, string? projectPath)> FindAssetsFiles(string path, ILoggerWithColor logger)
+        {
+            var extension = Path.GetExtension(path);
+            if (string.Equals(".json", extension, StringComparison.OrdinalIgnoreCase))
+            {
+                yield return (path, null);
+                yield break;
+            }
+
+            var projectPaths = MSBuildAPIUtility.GetListOfProjectsFromPathArgument(path);
+            foreach (string projectPath in projectPaths.NoAllocEnumerate())
+            {
+                Project project = MSBuildAPIUtility.GetProject(projectPath);
+                try
+                {
+                    string usingNetSdk = project.GetPropertyValue("UsingMicrosoftNETSdk");
+                    if (bool.TryParse(usingNetSdk, out bool b) && b)
+                    {
+                        if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
                         {
-                            whyCommandArgs.Logger.LogMinimal(
+                            logger.LogError(
                                 string.Format(
-                                    Strings.WhyCommand_Message_DependencyGraphsFoundInProject,
-                                    assetsFile.PackageSpec.Name,
-                                    targetPackage));
+                                    CultureInfo.CurrentCulture,
+                                    Strings.Error_NotPRProject,
+                                    project.FullPath));
 
-                            DependencyGraphPrinter.PrintAllDependencyGraphs(dependencyGraphPerFramework, targetPackage, whyCommandArgs.Logger);
+                            continue;
                         }
-                        else
+
+                        string? assetsFilePath = project.GetPropertyValue(ProjectAssetsFile);
+                        if (string.IsNullOrEmpty(assetsFilePath))
```

Comment:
Test `NuGet.XPlat.FuncTest.XPlatWhyTests.WhyCommand_ProjectDidNotRunRestore_Fails` is failing because the wrong message is being output. Debugging, this property does not have a value, but for the life of me, I can't figure out why.

If I go to the command line, `dotnet msbuild -getProperty:ProjectAssetsFile -bl` also returns an empty string (unless I accidentally restore first, but delete the `obj` directory and it's empty again), but `dotnet build --no-restore -bl` shows in the binlog that the property is set with the correct value. If I delete the csproj that the test creates and use `dotnet new classlib`, then this weied behaviour doesn't happen. But looking at the binlogs, I just can't figure out why our test project behaves like this.

Reply:
Retrying `dotnet new classlib` `dotnet msbuild -getProperty:ProjectAssetsFile` returns an empty string, so I think it is an issue with the SDK props/targets.  So, I changed the warning message to match the "you need to restore this project" message.